src/gc_gl.c

```diff
@@ -738,8 +749,15 @@ void glVertex2f(GLfloat x, GLfloat y)
 
 void glVertex3f(GLfloat x, GLfloat y, GLfloat z)
 {
-    if (glparamstate.imm_mode.current_numverts >= NUM_VERTS_IM)
-        return;
+    if (glparamstate.imm_mode.current_numverts >= glparamstate.imm_mode.current_vertices_size) {
+        if (!glparamstate.imm_mode.current_vertices) return;
+        int new_size = glparamstate.imm_mode.current_vertices_size * 2;
```

Comment:
Not sure how keen I am on doubling every time. Have you done any profiling of how many vertices you end up with at peak usage? I can see doubling when current_vertices_size is small but maybe cap the increment somewhere? Is there a benefit to starting small?

Reply:
I'm afraid it really depends on the application. I noticed the problem by running [this code](https://rosettagit.org/drafts/boids-c/), and the maximum there is 162. Starting small makes sense, because we don't know how the API will be used and what is most important for the client: maybe it has objcts with a low number of vertices but high-definition textures, so it would rather save memory for that. Note that each vertex takes up quite a bit of memory (it's 12 floats, so 48 bytes).

I do agree that capping the increment makes sense. Maybe you double until we get to 256, after which we just add 256 every time. I'll do like this.

Reply:
Done!